R/test-directory-c.R

```diff
@@ -90,7 +88,7 @@ test_dir_parallel <- function(path,
     TESTTHAT_DIR = testthat_dir
   ))
 
-  paths <- find_test_scripts(path, filter, ...)
+  paths <- sort(find_test_scripts(path, filter, ...))
```

Comment:
One way to 'solve' your slowest tests is the last one problem would be to randomize the order the test files are run in.

This would also more quickly smoke out hidden dependencies between test files.

I think the big drawback would be the order of test outputs would then no longer be deterministic, which probably makes it a non-starter.

Reply:
Yeah, I thought about that. Maybe eventually we'll have a config parameter for this, but I think in general it would be nice if testthat remembered the last test results. E.g. you could re-run the failing tests easily. And if we remember the previous results, then we can just order the files according the previous results.

Reply:
Btw. I don't think a random order is a non-starter. We are used to seeing the lines of successful test files, but honestly, while this is heart-warming, it is not very informative. I'd rather see a very short summary of all successes, and then the execution order does not matter that much.

Reply:
I guess we'll need to experiment with this a bit....

Reply:
Actually, maybe we should randomize the order by default. Because otherwise the interdependence might not come up until much later, potentially on CRAN. Just like it came up for testthat itself on GHA...

Reply:
I am leaning towards a randomized order on the CI now, and alphabetical order locally, for a better interactive experience. But it should be all configurable.